iam.tf

```diff
@@ -1,30 +1,115 @@
 locals {
-  iam_map = var.policy_bindings == null ? { for iam in var.iam : iam.role => iam } : {}
+  # filter all objects that define a single role
+  iam_role = [for iam in var.iam : iam if can(iam.role)]
 
-  policy_bindings = var.policy_bindings != null ? {
-    iam_policy = {
-      policy_bindings = var.policy_bindings
-    }
-  } : {}
+  # filter all objects that define multiple roles and expand them to single roles
+  iam_roles = flatten([for iam in var.iam :
+    [for role in iam.roles : merge(iam, { role = role })] if can(iam.roles)
+  ])
 
-  iam_output = [local.policy_bindings, local.iam_map]
+  # we allow to define role and roles in the same object and concatenate them
+  # each role can only be sepcified exactly once
+  iam = concat(local.iam_role, local.iam_roles)
 
-  iam_output_index = var.policy_bindings != null ? 0 : 1
+  iam_map = { for idx, iam in local.iam :
+    try(iam._key, iam.role) => idx
+  }
```

Comment:
please also backport the fix used in https://github.com/mineiros-io/terraform-google-service-account/pull/35 here...

Reply:
sure chief

Reply:
0-day fixes ;)

Reply:
Done bud